modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,367 @@
+<?php
+/**
+ * Module Name: Dominant Color
+ * Description: Adds support to store dominant-color for an image and create a placeholder background with that color.
+ * Experimental: Yes
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Add the dominant color metadata to the attachment.
+ *
+ * @since n.e.x.t
+ *
+ * @param array $metadata The attachment metadata.
+ * @param int   $attachment_id The attachment ID.
+ * @return array $metadata The attachment metadata.
+ */
+function dominant_color_metadata( $metadata, $attachment_id ) {
+	if ( ! wp_attachment_is_image( $attachment_id ) ) {
+
+		return $metadata;
+	}
+
+	$dominant_color = dominant_color_get( $attachment_id );
+
+	if ( ! empty( $dominant_color ) ) {
+		$metadata['dominant_color'] = $dominant_color;
+	}
+
+	$has_transparency = dominant_color_get_has_transparency( $attachment_id );
+
+	if ( ! empty( $has_transparency ) ) {
```

Comment:
This needs to be set unless it's `null`, based on how `dominant_color_get_has_transparency()` works.
```suggestion
	if ( is_bool( $has_transparency ) ) {
```

I still think it would be cleaner to return a `WP_Error` instead of `null` when it cannot be detected.